src/test/scala/mesosphere/marathon/state/TaskFailureRepositoryTest.scala

```diff
@@ -0,0 +1,74 @@
+package mesosphere.marathon.state
+
+import com.codahale.metrics.MetricRegistry
+import mesosphere.marathon.MarathonSpec
+import mesosphere.marathon.metrics.Metrics
+import mesosphere.util.state.memory.InMemoryStore
+import org.scalatest.{ Matchers, GivenWhenThen }
+
+class TaskFailureRepositoryTest extends MarathonSpec with GivenWhenThen with Matchers {
+  import TaskFailureTestHelper.taskFailure
+  import org.scalatest.concurrent.ScalaFutures._
+
+  test("store and fetch") {
+    Given("an empty repo")
+    val f = new Fixture
+
+    When("we store a taskFailure")
+    f.taskFailureRepo.store(PathId("/some/app"), taskFailure).futureValue
+
+    And("fetch it")
+    val readFailure = f.taskFailureRepo.current(PathId("/some/app")).futureValue
+
+    Then("the resulting failure is the one we stored")
+    readFailure should be(Some(taskFailure))
+  }
+
+  test("the last store wins") {
+    Given("an empty repo")
+    val f = new Fixture
+
+    When("we store a taskFailure")
+    f.taskFailureRepo.store(PathId("/some/app"), taskFailure).futureValue
+
+    And("another one for the same app")
+    val anotherTaskFailure = taskFailure.copy(message = "Something else")
+    f.taskFailureRepo.store(PathId("/some/app"), anotherTaskFailure).futureValue
+
+    And("fetch it")
+    val readFailure = f.taskFailureRepo.current(PathId("/some/app")).futureValue
+
+    Then("the resulting failure is the one we stored LAST")
+    readFailure should be(Some(anotherTaskFailure))
+  }
+
+  test("expunge works") {
+    Given("an empty repo")
+    val f = new Fixture
+
+    When("we store a taskFailure")
+    f.taskFailureRepo.store(PathId("/some/app"), taskFailure).futureValue
+
+    And("expunge it again")
+    f.taskFailureRepo.expunge(PathId("/some/app")).futureValue
+
+    And("fetch it")
+    val readFailure = f.taskFailureRepo.current(PathId("/some/app")).futureValue
+
+    Then("the result is None")
+    readFailure should be(None)
+  }
+
+  class Fixture {
+    lazy val inMemoryStore = new InMemoryStore()
```

Comment:
It's fast. You do not need zookeeper or mesos. So why not always run it?

What do you think?